src/messaging/ExchangeACL.h

```diff
@@ -0,0 +1,100 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/util/basic-types.h>
+#include <support/ReturnMacros.h>
+#include <transport/AdminPairingTable.h>
+
+namespace chip {
+namespace Messaging {
+
+/**
+ * @brief Defines a class that encapsulates ACL target information (cluster and endpoint ID).
+ *        The class can be extended to add other parameters to the ACL target.
+ */
+class DLL_EXPORT ACLTarget
+{
+public:
+    ACLTarget(ClusterId clusterId, EndpointId endpoint) : mClusterId(clusterId), mEndpoint(endpoint) {}
+    virtual ~ACLTarget() {}
+
+    ClusterId GetClusterId() { return mClusterId; }
+    EndpointId GetEndpointId() { return mEndpoint; }
+
+private:
+    ClusterId mClusterId;
+    EndpointId mEndpoint;
+};
+
+/**
+ * @brief Defines the common interface for PASE/CASE/GroupID based ACL permissions check.
+ */
+class DLL_EXPORT ExchangeACL
+{
+public:
+    enum class PermissionLevel
+    {
+        kNone,
+        kView,
+        kOperate,
+        kManage,
+        kAdminister,
+    };
+
+    virtual ~ExchangeACL() {}
+
+    /**
+     * @brief
+     *   Check access permissions for the message received from subject (sender) that
+     *   are trying to access the target (e.g. cluster and endpoint).
+     *
+     * @param subject    Typically the source Node id in the message
+     * @param target     The target of the message (i.e. cluster and endpoint)
+     *
+     * @return Permissions granted by the configured ACLs
+     */
+    virtual PermissionLevel GetPermissionLevel(NodeId subject, const ACLTarget & target) = 0;
```

Comment:
NodeId would be part of the ACL:
  - exchanges are per node-id (or group)
  - groups have no nodeid
  
I think a set/get nodeID should be part  of CASE ACL.
PASEACL and GroupACL will not have one

Reply:
@andy31415 I added a base class for subject in the latest commit. Let me know if that semantic looks ok.